runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/jobsubmission/JobInvocation.java

```diff
@@ -93,6 +95,7 @@ public void onSuccess(@Nullable PipelineResult pipelineResult) {
               switch (pipelineResult.getState()) {
                 case DONE:
                   setState(Enum.DONE);
+                  metrics = pipelineResult.portableMetrics();
```

Comment:
I notice all Flink results are `DONE`:
https://github.com/apache/beam/blob/949c972ec837cb9350532ad4b5c0f261faa3f9df/runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkRunnerResult.java#L49-L51
 but maybe we should populate metrics for the other cases if other runners handle this differently.

Also, could we consider populating metrics in `onFailure`?

Reply:
What do you think of populating metrics only when the state `isTerminal()`? 

so:

```
if(pipelineResult.getState().isTerminal()) {
    metrics = pipelineResult.portableMetrics();
}
```

> Also, could we consider populating metrics in onFailure?

I don't think it is possible - notice that we only have access to `Throwable` in `onFaulure()` method so no easy way to get monitoring infos in here (we need pipelineResult for that).

Reply:
Sure, looks good :+1: